code/obj/machinery/manufacturer.dm

```diff
@@ -1469,6 +1495,18 @@
 				src.timeleft += rand(2,6)
 				src.timeleft *= 1.5
 			src.timeleft /= src.speed
+
+		if(src.fabdisk)
+			if(src.queue[1] in src.disk)
+				var/obj/item/fabdisk/FD = src.fabdisk
+				if(FD.fablimit == 0)
+					src.mode = "halt"
+					src.error = "This ManuDisk is unable to operate further."
+					src.queue = list()
+					return
+				else
+					FD.fablimit -= 1
```

Comment:
might be worth changing logic so fabdisk charges are only considered/consumed if the recipe is otherwise not in the fab

Reply:
Considered it, theres no point in having a limit on something that you can get infinitely from a fab. The logic working like that shouldnt cause any issues.